execution/execution.proto

```diff
@@ -0,0 +1,88 @@
+syntax = "proto3";
+
+package execution;
+
+import "types/types.proto";
+
+option go_package = "./execution;execution";
+
+enum ValidationStatus {
+    Success = 0;      // State transition simulation is successful.
+    InvalidChain = 1; // State transition simulation is Unsuccessful.
+    TooFarAway = 2;   // Chain hash is too far away from current chain head and unfeasible to validate.
+    MissingSegment = 3; // Chain segments are missing.
+}
+
+message ForkChoiceReceipt {
+    bool success = 0; // Forkchoice is either successful or unsuccessful.
+}
+
+// Result we receive after validation
+message ValidationReceipt {
+    ValidationStatus validationStatus = 0;
+    optional types.H256 missingHash = 1; // The missing hash, in case we receive MissingSegment so that we can reverse download it.
+};
+
+message IsCanonicalResponse {
+    bool canonical = 0; // Whether hash is canonical or not.
+}
+
+// Header is an header for execution
+message Header {
+  types.H256 parentHash = 1;
+  types.H160 coinbase = 2;
+  types.H256 stateRoot = 3;
+  types.H256 receiptRoot = 4;
+  types.H2048 logsBloom = 5;
+  types.H256 mixDigest = 6;
+  uint64 blockNumber = 7;
+  uint64 gasLimit = 8;
+  uint64 gasUsed = 9;
+  uint64 timestamp = 10;
+  bytes extraData = 11;
+  types.H256 baseFeePerGas = 12;
+  types.H256 blockHash = 13; // We keep this so that we can validate it
+}
+
+// Body is a block body for execution
+message BlockBody {
+  types.H256 blockHash = 1;
+  uint64 blockNumber = 2;
+  // Raw transactions in byte format.
+  repeated bytes transactions = 3;
```

Comment:
Perhaps "uncles" here too?

Reply:
well whatever adding them wont kill me

Reply:
They are needed to support downloading older PoW blocks